src/main/scala/fi/oph/koski/validation/KoskiValidator.scala

```diff
@@ -927,11 +928,43 @@ class KoskiValidator(
       }
   }
 
+  private def validateSuoritustenLuokkaAsteet(opiskeluoikeus: KoskeenTallennettavaOpiskeluoikeus): HttpStatus = {
+    opiskeluoikeus match {
+      case oo: PerusopetuksenOpiskeluoikeus => {
+        val nuortenPerusopetuksenErityinenTutkintoSuoritukset = oo.suoritukset.filter(_.isInstanceOf[NuortenPerusopetuksenOppiaineenOppimääränSuoritus]).asInstanceOf[List[NuortenPerusopetuksenOppiaineenOppimääränSuoritus]].filter(_.suoritustapa.koodiarvo == "erityinentutkinto")
```

Comment:
collect + case on scalampaa ja välttää asInstanceOf:in, eli:
`oo.suoritukset.collect( { case s: NuortenPerusopetuksenOppiaineenOppimääränSuoritus => s })`